### PR TITLE
Lvnod-74-Profile dropdown 

### PR DESCRIPTION
**Pull request by yuliannakorol**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/browse/LVNOD-74

## Code reviewers

- [x] @yarvanya
- [ ] @yurii-kokhalevych
- [ ] @denisbodnar

### Second Level Review

- [ ] @lhalam
- [ ] @4asof4ik

## Summary of change
- Created Profile drop-down component for navigation bar
- Created Profile reducers
- Created test Profile container (will be developed in another task and PR)

## CHECK LIST
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR

### CODE REVIEW

File: app/src/components/Navigation/Navigation.js [hunk 1]
```diff
@@ -1,9 +1,10 @@
 import React from 'react';
-import { Navbar, Nav, NavItem, Button } from 'react-bootstrap';
+import { Navbar, Nav, NavItem } from 'react-bootstrap';
 import { LinkContainer } from 'react-router-bootstrap';
 import { connect } from 'react-redux';
 import { bindActionCreators } from 'redux';
 import * as loginActions from '../../redux/Login';
+import {ProfileDropDown} from './../';
```
Comment (4asof4ik) [inline app/src/components/Navigation/Navigation.js]: spaces after and before {}

File: app/src/components/ProfileDropDown/ProfileDropDown.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import { LinkContainer } from 'react-router-bootstrap';
+import { Link } from 'react-router';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import * as loginActions from '../../redux/Login';
+import * as profileActions from '../../redux/ProfileReducers';
+import './profileDropDown.scss';
+
+class ProfileDropDown extends React.Component {
+
```
Comment (4asof4ik) [inline app/src/components/ProfileDropDown/ProfileDropDown.js]: here add constructor with state profile

File: app/src/components/ProfileDropDown/ProfileDropDown.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  componentWillMount() {
+    this.props.actions.loginActions.checkToken();
+    this.props.actions.profileActions.retrieveProfile(this.props.userId);
+  }
+
+  render() {
+    const profile = this.props.profile;
+    return (
+      <div >
+        <NavDropdown style={{display: 'inline-block'}} eventKey={5} id='basic-nav-dropdown'
```
Comment (4asof4ik) [inline app/src/components/ProfileDropDown/ProfileDropDown.js]: style={{display: 'inline-block'}} what 0_o

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

**Review by yarvanya — DISMISSED** [review]

**Review by yarvanya — DISMISSED** [review]

**Review by yurii-kokhalevych — DISMISSED** [review]

**Review by denisbodnar — DISMISSED** [review]

File: app/src/redux/reducer.js [hunk 4]
```diff
@@ -3,11 +3,13 @@ import { routerReducer } from 'react-router-redux';
 import signup from './Signup';
 import login from './Login';
 import event from './eventReducers';
+import profile from './ProfileReducers';
```
Comment (4asof4ik) [inline app/src/redux/reducer.js]: rename all reducer file with camelCase

File: app/src/components/ProfileDropDown/ProfileDropDown.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import * as loginActions from '../../redux/Login';
+import * as profileActions from '../../redux/ProfileReducers';
+import './profileDropDown.scss';
+
+class ProfileDropDown extends React.Component {
+
+  componentWillMount() {
+    this.props.actions.loginActions.checkToken();
```
Comment (4asof4ik) [inline app/src/components/ProfileDropDown/ProfileDropDown.js]: why check token here?

File: app/src/containers/Profile/Profile.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+        <h1>My Profile Info</h1>
+        <p>First Name: {profile.firstName}</p>
+        <p>Last Name: {profile.lastName}</p>
+      </div>
+    );
+  }
+}
+
+const mapStateToProps = state => ({
+  profile: state.profile.profile,
```
Comment (4asof4ik) [inline app/src/containers/Profile/Profile.js]: profile.profile ?

File: app/src/containers/Profile/Profile.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+  userId: state.login.userId
```
Comment (4asof4ik) [inline app/src/containers/Profile/Profile.js]: you have userId in sesion storage

File: app/src/redux/ProfileReducers.js [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+  return dispatch => {
+    dispatch(retrieveProfilerRequest());
+    return request()
+      .get(API.HOST + API.PORT + '/api/user/' + userId)
+      .end((err, res) => {
+        if (err) {
+          dispatch({
+            type: RETRIEVE_PROFILE_FAILURE,
+            payload: err
```
Comment (4asof4ik) [inline app/src/redux/ProfileReducers.js]: error

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

**Review by yurii-kokhalevych — DISMISSED** [review]

**Review by yarvanya — DISMISSED** [review]

File: app/src/components/ProfileDropDown/ProfileDropDown.js [hunk 9]
```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { Navbar, Nav, NavItem, NavDropdown, MenuItem, Image } from 'react-bootstrap';
+import { LinkContainer } from 'react-router-bootstrap';
+import { Link } from 'react-router';
```
Comment (4asof4ik) [inline app/src/components/ProfileDropDown/ProfileDropDown.js]: where you use this?

File: app/src/components/ProfileDropDown/ProfileDropDown.js [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+import './profileDropDown.scss';
+
+class ProfileDropDown extends React.Component {
+
+  componentWillMount() {
+    this.props.actions.profileActions.retrieveProfile(sessionStorage.getItem('userId'));
+  }
+
+  render() {
+    const profile = this.props.profile;
```
Comment (4asof4ik) [inline app/src/components/ProfileDropDown/ProfileDropDown.js]: { profile, action } = this.props

File: app/src/components/ProfileDropDown/ProfileDropDown.js [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+    return (
+      <div>
+        <NavDropdown className='profileDropDown' eventKey={5} id='basic-nav-dropdown'
+          title={profile.firstName + ' ' + profile.lastName}>
+          <LinkContainer to='/profile'>
+            <MenuItem eventKey={5.1}>Edit Profile</MenuItem>
+          </LinkContainer>
+          <MenuItem eventKey={5.2} onClick={this.props.actions.loginActions.logout}>Log Out</MenuItem>
```
Comment (4asof4ik) [inline app/src/components/ProfileDropDown/ProfileDropDown.js]: actions.loginActions.logout

File: app/src/containers/Profile/Profile.js [hunk 12]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { Link } from 'react-router';
+import { connect } from 'react-redux';
+
+class Profile extends React.Component {
+
+  render() {
+    const profile = this.props.profile;
```
Comment (4asof4ik) [inline app/src/containers/Profile/Profile.js]: look above

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

**Review by 4asof4ik — APPROVED** [review]

**Review by yarvanya — APPROVED** [review]

**Pull request merged**